client/src/pages/lessons/edit.tsx

```diff
@@ -396,692 +360,606 @@ const LessonEdit = (props: {
+        intro: e.target.value || "",
+      },
+      true
+    );
+  };
+
+  const handleQuestion = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setLesson(
+      {
+        ...(lessonUnderEdit.lesson || newLesson),
+        question: e.target.value || "",
+      },
+      true
+    );
+  };
+
+  const handleVideoStart = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setLesson(
+      {
+        ...(lessonUnderEdit.lesson || newLesson),
+        media: {
+          url: lessonUnderEdit.lesson?.media?.url || "",
+          type: MediaType.VIDEO,
+          props: copyAndSetProp(
+            (lessonUnderEdit.lesson || newLesson).media?.props || [],
+            {
+              name: "start",
+              value: String(parseFloat(e.target.value) || 0) || "",
+            }
+          ),
+        },
+      },
+      true
+    );
+  };
+
+  const handleVideoEnd = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setLesson(
+      {
+        ...(lessonUnderEdit.lesson || newLesson),
+        media: {
+          url: lessonUnderEdit.lesson?.media?.url || "",
+          type: MediaType.VIDEO,
+          props: copyAndSetProp(
+            (lessonUnderEdit.lesson || newLesson).media?.props || [],
+            {
+              name: "end",
+              value:
+                String(parseFloat(e.target.value) || Number.MAX_SAFE_INTEGER) ||
+                "",
+            }
+          ),
+        },
+      },
+      true
+    );
+  };
+
+  const handleVideoURL = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setLesson(
+      {
+        ...(lessonUnderEdit.lesson || newLesson),
+        media: {
+          ...(lessonUnderEdit.lesson || newLesson).media,
+          type: MediaType.VIDEO,
+          url: (e.target.value as string) || "",
+        },
+      },
+      true
+    );
+  };
+
+  const handleImageURL = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setLesson(
+      {
+        ...(lessonUnderEdit.lesson || newLesson),
+        media: {
+          ...(lessonUnderEdit.lesson || newLesson).media,
+          type: MediaType.IMAGE,
+          url: (e.target.value as string) || "",
+        },
+      },
+      true
+    );
+  };
+
+  const handleLessonId = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setLesson(
+      {
+        ...(lessonUnderEdit.lesson || newLesson),
+        lessonId: e.target.value || "",
+      },
+      true
+    );
+  };
+
+  const handleDialogCategory = (e: SelectChangeEvent<string>) => {
+    setLesson(
+      {
+        ...(lessonUnderEdit.lesson || newLesson),
+        dialogCategory: (e.target.value as string) || "",
+      },
+      true
+    );
+  };
+
+  const handleClassifierArch = (e: SelectChangeEvent<string>) => {
+    setLesson(
+      {
+        ...(lessonUnderEdit.lesson || newLesson),
+        arch: (e.target.value as string) || DEFAULT_CLASSIFIER_ARCHITECTURE,
+      },
+      true
+    );
+  };
+
+  const AdvancedFeatures = () => {
```

Comment:
The AdvancedFeatures component should be moved into it's own file. Overall I think the components in this file could be modularized a bit more